### PR TITLE
Various CI improvements

### PR DESCRIPTION
**Pull request by pmav99**
There are various changes in this PR. The main points are:

- the addition of VCR cassettes for the slow COOPs tests which makes the test-suite quite a bit faster (at least for those of us with relatively slow internet speeds :P )
- uses the pre-commit action for running the linters. This way we will have the exact same setup both on our workstations and on CI. No longer problems with different versions 
- Adds some poetry-specific pre-commit hooks. More specifically, we now check that `pyproject.toml`, `poetry.lock` and `requirements.txt` are kept in sync.

### CODE REVIEW

File: .github/workflows/run_tests.yml [hunk 1]
```diff
@@ -25,51 +25,43 @@ jobs:
...
-        run: black --check .
-      - name: lint with prospector
-        run: make lint
-      - name: lint with mypy
-        run: make mypy
-  test:
-    needs: lint
-    name: test
+      - name: Install (dev) dependencies
+        run: python -mpip install -r requirements/requirements-dev.txt
```
Comment (zacharyburnett) [inline .github/workflows/run_tests.yml:29]: Should this be `-m pip`

File: .github/workflows/run_tests.yml [hunk 2]
```diff
@@ -25,51 +25,43 @@ jobs:
...
-        run: pip install . pytest pytest-cov
-      #----------------------------------------------
-      # Run tests
-      #----------------------------------------------
-      - name: Run tests with coverage
-        run: |
-          pytest tests/
-          coverage report
+      - name: Install (dev) dependencies
+        run: python -mpip install -r requirements/requirements-dev.txt
```
Comment (zacharyburnett) [inline .github/workflows/run_tests.yml:63]: Same here

File: pyproject.toml [hunk 3]
```diff
@@ -53,7 +56,7 @@ pytest-recording = "^0.12.0"
 types-requests = "^2.26"
 
 [tool.poetry.extras]
-linting = ["black", "mypy", "prospector"]
+linting = ["mypy", "prospector"]
```
Comment (zacharyburnett) [inline pyproject.toml:59]: Why this change? Shouldn't we keep black in the extras, or is that covered in the precommit?

**Review by zacharyburnett — CHANGES_REQUESTED** [review]
Looks good to me, after changing `-mpip` I think it will be good to merge. I did not think about internet connectivity and running the tests, so I appreciate this addition of the cassettes!

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (pmav99) [reply .github/workflows/run_tests.yml:29]: You mean `-m pip` vs `-mpip`? They are the same thing.

File: pyproject.toml (hunk 3 above)
Comment (pmav99) [reply pyproject.toml:59]: My thought was that since we install black via pre-commit there is no reason to install it in the environment, too. This way we don't need to reference `black` in multiple files (`pyproject.toml` vs `.pre-commit-config.yaml`) and keep versions in sync. So, since `black` was no longer referenced in `pyproject.toml` I removed it from the extras, too.

Comment (pmav99) [reply pyproject.toml:59]: PS. You probably already know this, but just in case, black has introduced a forward-compatible policy:
> Starting in 2022, the formatting output will be stable for the releases made in the same year (other than unintentional bugs). 
Therefore versions `2022.3` up to `2022.12` should only contain bugfixes. Breaking changes will come in `2023.*`
[source](https://black.readthedocs.io/en/stable/faq.html#how-stable-is-black-s-style)

File: .github/workflows/run_tests.yml (hunk 1 above)
Comment (pmav99) [reply .github/workflows/run_tests.yml]: Fixed: https://github.com/oceanmodeling/searvey/pull/25/commits/e28e87e1c64399f47fd2669aa51cea5008b2d7d5

**Review by zacharyburnett — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (zacharyburnett) [reply pyproject.toml:59]: I didn't know that, that's great! That will make things a lot easier with pinning versioning
